shared/git_monkey_patches.rb

```diff
@@ -0,0 +1,16 @@
+module Git
+   class Lib
+    # Monkey patch ls_files until https://github.com/ruby-git/ruby-git/pull/320 is resolved
+    def ls_files(location=nil)
+      location ||= '.'
+      hsh = {}
+      command_lines('ls-files', ['--stage', location]).each do |line|
+        (info, file) = line.split("\t")
+        (mode, sha, stage) = info.split
+        file = eval(file) if file =~ /^\".*\"$/ # This takes care of quoted strings returned from git
+        hsh[file] = {:path => file, :mode_index => mode, :sha_index => sha, :stage => stage}
+      end
+      hsh
```

Comment:
explicit return statement 😄

Reply:
Also let's comment where the code came from 👍